crates/rethnet_evm/benches/util.rs

```diff
@@ -86,61 +129,79 @@ mod config {
     ];
 
     pub const STORAGE_SCALES: [u64; 4] = [1, 10, 100, 1000];
+
+    pub const SNAPSHOT_SCALES: [u64; 4] = [1, 10, 100, 1000];
 }
 
-pub use config::STORAGE_SCALES;
 use config::*;
+pub use config::{SNAPSHOT_SCALES, STORAGE_SCALES};
 
 pub fn bench_sync_state_method<O, R, Prep>(
     c: &mut Criterion,
     method_name: &str,
     mut prep: Prep,
     mut method_invocation: R,
     storage_scales: &[u64],
+    snapshot_scales: &[u64],
 ) where
-    R: FnMut(Box<dyn SyncState<StateError>>, u64) -> O,
+    R: FnMut(Box<dyn SyncState<StateError>>, u64, &Vec<B256>, &Vec<B256>) -> O,
     Prep: FnMut(&mut dyn SyncState<StateError>, u64),
 {
     let mut group = c.benchmark_group(method_name);
-    for accounts_per_checkpoint in CHECKPOINT_SCALES.iter() {
+    for number_of_checkpoints in CHECKPOINT_SCALES.iter() {
         for number_of_accounts in ADDRESS_SCALES.iter() {
             for storage_slots_per_account in storage_scales.iter() {
-                let mut rethnet_states = RethnetStates::default();
-                rethnet_states.fill(
-                    *number_of_accounts,
-                    *accounts_per_checkpoint,
-                    *storage_slots_per_account,
-                );
+                for number_of_snapshots in snapshot_scales.iter() {
+                    let mut rethnet_states = RethnetStates::default();
+                    rethnet_states.fill(
+                        *number_of_accounts,
+                        *number_of_checkpoints,
+                        *number_of_snapshots,
+                        *storage_slots_per_account,
+                    );
 
-                for (label, state_factory) in rethnet_states.make_clone_factories().into_iter() {
-                    group.bench_with_input(
-                        BenchmarkId::new(
-                            format!(
-                                "{},{} account(s) per checkpoint, {} storage slots per account",
-                                label, *accounts_per_checkpoint, *storage_slots_per_account
+                    for (label, state_factory, checkpoints, snapshots) in
+                        rethnet_states.make_state_refs().into_iter()
+                    {
+                        group.bench_with_input(
+                            BenchmarkId::new(
+                                format!(
+                                    "{},{} chkpts,{} slots per acct,{} snapshots",
```

Comment:
no. of accounts?

Reply:
i removed "number of accounts" from the "function name" because "number of accounts" is the value we're using as the "input" to that function, so that value is automatically appended to every permutation of the "benchmark name"